src/ethereum_test_execution/transaction_post.py

```diff
@@ -51,11 +59,38 @@ def execute(
                 for transaction in signed_txs:
                     if transaction.error is None:
                         eth_rpc.send_wait_transaction(transaction)
+                        all_tx_hashes.append(transaction.hash)
                     else:
                         with pytest.raises(SendTransactionExceptionError):
                             eth_rpc.send_transaction(transaction)
             else:
                 eth_rpc.send_wait_transactions(signed_txs)
+                all_tx_hashes.extend([tx.hash for tx in signed_txs])
+
+        # Perform gas validation if required for benchmarking
+        # Ensures benchmark tests consume exactly the expected gas
+        if not self.skip_gas_used_validation:
+            # Use expected_benchmark_gas_used if set, else gas_benchmark_value
+            expected_gas = self.expected_benchmark_gas_used
+            if expected_gas is None:
+                expected_gas = self.gas_benchmark_value
```

Comment:
This is not required after the `expected_benchmark_gas_used` value is configured. It would have default value to `gas_benchmark_value`, but you might encounter some typing issue, it could be resolved as


```suggestion
            assert expected_gas is not None, "expected benchmark gas value is not set"
```